WPFHospitalEditor/EquipmentRelocation.xaml.cs

```diff
@@ -0,0 +1,167 @@
+        private DateTime startDate;
+        private DateTime endDate;
+
+        public EquipmentRelocation(string relocationEquipmentName, int roomId)
+        {
+            SearchResultDialog.selectedObjectId = -1;
+            InitializeComponent();
+            equipmentName.Text += relocationEquipmentName;
+            this.destinationRoomId = roomId;
+            this.relocationEquipmentName = relocationEquipmentName;
+            FillObservableCollection();
+            SetRoomsComboBox();
+            this.DataContext = this;
+        }
+
+        private void CloseClick(object sender, RoutedEventArgs e)
+        {
+            this.Close();
+        }
+
+        private void SetRoomsComboBox()
+        {
+            List<Room> rooms = roomServerController.getRoomsByEquipmentType(relocationEquipmentName).ToList();
+            foreach (Room room in rooms)
+            {
+                if (room.Id != destinationRoomId)
+                    roomSearchComboBox.Items.Add(room.Id);
+            }
+        }
+
+        private void RelocateEquipment(object sender, RoutedEventArgs e)
+        {
+            startDate =
+                DateTime.ParseExact(startDatePicker.SelectedDate.Value.ToString("MM/dd/yyyy")
+                + " " + StartTime.Text, "MM/dd/yyyy HH:mm", null);
+            endDate =
+                DateTime.ParseExact(startDatePicker.SelectedDate.Value.ToString("MM/dd/yyyy")
+                + " " + EndTime.Text, "MM/dd/yyyy HH:mm", null);
+            TimeInterval timeInterval = new TimeInterval(startDate, endDate);
+
+            if (AmountIsValid() && timeInterval.IsValid())
+            {
+                EquipmentRelocationDto eqRelDto = CreateEquipmentRelocationDto(timeInterval);
+                List<int> unavailableRooms = roomServerController.GetUnavailableRoomsIdsInTimeInterval(eqRelDto).ToList();
+                if (unavailableRooms.Count > 0)
+                {
+                    ShowAlternativeRelocationAppointments(unavailableRooms);
+                }
+                else
+                {
+                    ScheduleRelocation(eqRelDto);
+                }
+            }
+
+        }
+
+        private void ShowAlternativeRelocationAppointments(List<int> unavailableRooms)
+        {
+            AlternativeRelocationAppointments newWindow =
+            new AlternativeRelocationAppointments(unavailableRooms[0], this);
+            newWindow.Show();
+        }
+
+        private void ScheduleRelocation(EquipmentRelocationDto eqRelDto)
+        {
+
+            List<int> doctors = doctorServerController.GetDoctorsByRoomsAndShifts(eqRelDto).ToList();
+            foreach (int doctorId in doctors)
+            {
+                startDate = eqRelDto.TimeInterval.Start;
+                while (startDate < endDate)
+                {
+                    examinationServerController.ScheduleExamination(startDate, doctorId, 2);
+                    startDate = startDate.AddMinutes(30);
+                }
```

Comment:
In relocation, scheduling exam for patient with ID 2 is little scary for that patient.

Reply:
Fixed.